lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -710,9 +710,20 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
 
   SILBuilderWithScope Builder(SemanticsCall);
   auto &ValLowering = Builder.getTypeLowering(V->getType());
+
+  // In OSSA, the InsertPt is after V's definition and not before SemanticsCall
+  // Because we are creating copy_value in ossa, and the source may have been
+  // taken previously. So our insert point for copy_value is immediately after
+  // V, where we can be sure it is live.
+  auto InsertPt = V->getFunction()->hasOwnership()
+                      ? getInsertAfterPoint(V)
+                      : SemanticsCall->getIterator();
+  assert(InsertPt.hasValue());
   if (hasGetElementDirectResult()) {
-    ValLowering.emitCopyValue(Builder, SemanticsCall->getLoc(), V);
-    SemanticsCall->replaceAllUsesWith(V);
+    SILValue CopiedVal =
+        SILBuilderWithScope(InsertPt.getValue())
+            .emitCopyValueOperation(SemanticsCall->getLoc(), V);
+    SemanticsCall->replaceAllUsesWith(CopiedVal);
   } else {
     auto Dest = SemanticsCall->getArgument(0);
```

Comment:
I think you can remove this else-branch completely. It stems from a time where the specializations of getElement did return an indirect result, even for loadable types. This should not be the case anymore. You can just bail if !hasGetElementDirectResult().

Reply:
Fixed this.